src/core/border_router/routing_manager.cpp

```diff
@@ -619,48 +705,37 @@ void RoutingManager::EvaluateNat64Prefix(void)
 {
     OT_ASSERT(mIsRunning);
 
-    NetworkData::Iterator            iterator = NetworkData::kIteratorInit;
-    NetworkData::ExternalRouteConfig config;
-    Ip6::Prefix                      smallestNat64Prefix;
+    Ip6::Prefix                      nat64Prefix;
+    RoutePreference                  routePreference;
+    Error                            error;
+    NetworkData::ExternalRouteConfig preferredNat64PrefixConfig;
+    bool                             should_advertise;
 
     LogInfo("Evaluating NAT64 prefix");
 
-    smallestNat64Prefix.Clear();
-    while (Get<NetworkData::Leader>().GetNextExternalRoute(iterator, config) == kErrorNone)
-    {
-        const Ip6::Prefix &prefix = config.GetPrefix();
+    VerifyOrExit(GetNat64Prefix(nat64Prefix, &routePreference) == kErrorNone);
```

Comment:
```suggestion
    IgnoreError(GetNat64Prefix(nat64Prefix, &routePreference));
```
Impossible to fail here?

Reply:
Tried to use `IgnoreError` but it will cause build warning on `nat64Prefix`.